cpp/Dockerfile.alpine

```diff
@@ -0,0 +1,35 @@
+FROM alpine
+
+# install dependencies
+RUN apk add --no-cache -q \
+        autoconf \
+        bash \
+        bison \
+        boost-dev \
+        cmake \
+        flex \
+        g++ \
+        gcc \
+        git \
+        gzip \
+        lz4-dev \
+        make \
+        musl-dev \
+        ninja \
+        rapidjson-dev \
+        re2-dev \
+        snappy-dev \
+        wget \
+        zlib-dev \
+        zstd-dev
```

Comment:
@wesm without `ARROW_BUILD_TOOLCHAIN` set We vendor most of these 3rdparty dependencies, except [zlib](https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L825). What is Arrow's policy on resolving 3rdparty dependencies? Either conda(with toolchain set) or vendor almost everything?

Reply:
We haven't been investing energy in supporting packages provided by Linux distributions because the versions don't always line up (a lot of distros don't have Thrift 0.11 yet, for example). That's not to say that we can't, but we just aren't maintaining that path (instead leaving the ExternalProjects to do the work)